File: LeetCode/Python Solutions/Non-overlapping Intervals/nonover.py
from typing import List

class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        # If the list is empty, return 0
        if not intervals:
            return 0

        # Sort the intervals by their end time
        intervals.sort(key=lambda x: x[1])

        # Initialize the end time to the first interval's end time
        end = intervals[0][1]
        count = 0

        # Iterate through the intervals starting from the second one
        for i in range(1, len(intervals)):
            # If the current interval's start time is earlier than the last end time, increment the counter
            if intervals[i][0] < end:
                count += 1
            else:
                # Otherwise, update the end time
                end = intervals[i][1]

        return count

File: LeetCode/Java Solutions/Non-overlapping Intervals/nonover.java
import java.util.Arrays;
import java.util.Comparator;

class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        // If the array is null or has no elements, return 0
        if (intervals == null || intervals.length == 0) {
            return 0;
        }

        // Sort the intervals by their end time
        Arrays.sort(intervals, new Comparator<int[]>() {
            public int compare(int[] a, int[] b) {
                return a[1] - b[1];
            }
        });

        // Initialize the end time to the first interval's end time
        int end = intervals[0][1];
        int count = 0;

        // Iterate through the intervals starting from the second one
        for (int i = 1; i < intervals.length; i++) {
            // If the current interval's start time is earlier than the last end time, increment the counter
            if (intervals[i][0] < end) {
                count++;
            } else {
                // Otherwise, update the end time
                end = intervals[i][1];
            }
        }

        return count;
    }
}
